### PR TITLE
BL-853 Patch/auth0 fix

### PR DESCRIPTION
**Pull request by ElijahHopkin**
## Description
Removed Okta from all instances of FE code with some minor exceptions: SuperAdminForm.js, authenticateUser.js, and oktaConfig.js.  All are isolated components that don't connect with any other parts of the code so we left them for template guides when we build out Auth0 later.
Without Okta we had to rebuild the PrivateRoute.js to enable site navigation without authentication while we build Auth0.

Fixes # (issue)
Step one of transitioning from Okta to Auth0 per stakeholder request.

## Loom Video
https://www.loom.com/share/9e8b32b8169a4e1485b04b745dd80acf
## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have removed unnecessary comments/console logs from my code
- [x] I have checked my code and corrected any misspellings
- [x] No duplicate code left within changed files
- [x] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

**Review by MelodyLayne — APPROVED** [review]
The logic for leaving behind the remaining 4 places that okta exists is sound, I would also suggest that if those files are not in use that it couldn't hurt to create a separate folder to place them in. This way you still have a place to store the "guide" and are also able to finish cleaning out okta. Also I'd advocate for taking the okta references out of the package.json sooner rather than later. Otherwise, good work!

File: src/__tests__/MentorDashboard.test.js [hunk 1]
```diff
@@ -1,240 +1,3 @@
...
-//         name: 'Test004 User',
-//         email: 'llama004@maildrop.cc',
-//         preferred_username: 'llama004@maildrop.cc',
-//         role: 3,
-//       },
-//     };
-//   }),
-// }));
-
 test('Home Container no longer exists, this entire test needs to be revisited/refactored or deleted. It was originally for the Mentor Dashboard Test', () => {
```
Comment (ryan-hamblin) [inline src/__tests__/MentorDashboard.test.js:1]: add .skip to this line like so `test.skip()`. leaving in the console log is fine for.

File: src/auth/auth0ProviderWithHistory.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import { Auth0Provider } from '@auth0/auth0-react';
+
+const Auth0ProviderWithHistory = ({ children }) => {
+  const domain = process.env.REACT_APP_AUTH0_DOMAIN;
+  const clientId = process.env.REACT_APP_AUTH0_CLIENT_ID;
+
+  const history = useHistory();
+
+  const onRedirectCallback = appState => {
+    console.log(appState);
```
Comment (ryan-hamblin) [inline src/auth/auth0ProviderWithHistory.js:12]: let's remove this console.log(). I know we're still in dev mode here @paulstgermain might have wrote this and intended to remove it prior to production.

File: src/components/common/Loading.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+const loadingImg =
+  'https://cdn.auth0.com/blog/auth0-react-sample/assets/loading.svg';
```
Comment (ryan-hamblin) [inline src/components/common/Loading.js:3]: keep this as is for now, but in the future I'd love to meet with Swapnali and the Designers to get our very own fancy spinner that doesn't rely on `auth0` CDN to be delivered across the network.

File: src/components/common/PrivateRoute.js [hunk 4]
```diff
@@ -44,16 +33,15 @@ const PrivateRoute = ({
     allowRoles,
     redirect,
     dispatch,
-    oktaAuth,
-    authState,
     push,
   ]);
+  console.log(profile_id);
```
Comment (ryan-hamblin) [inline src/components/common/PrivateRoute.js:38]: I know this is WIP - Be sure to comment 'why' we just want to log this out for now.

File: src/components/pages/Navbar/Navbar.js [hunk 5]
```diff
@@ -8,27 +8,28 @@ import { UserOutlined, FormOutlined } from '@ant-design/icons';
...
+
   const openModal = () => setModal(true);
   const cancelOpen = () => setModal(false);
 
   const handleLogout = () => {
     setModal(false);
     localStorage.removeItem('role_id');
-    localStorage.removeItem('token');
-    oktaAuth.signOut();
+    localStorage.removeItem('AuthToken');
```
Comment (ryan-hamblin) [inline src/components/pages/Navbar/Navbar.js:31]: Awesome! now we're removing the Auth0 Token it seems :)

File: src/components/pages/Navbar/Navbar.js [hunk 6]
```diff
@@ -122,8 +123,10 @@ const Navbar = ({ isAuthenticated, userProfile, getProfile }) => {
               <div className="header_buttons">
                 <LoginButton />
                 <SignupButton />
+                <LogoutButton />
               </div>
             )}
+            {/* temporary logout button until private route is finished and when we can logout from dashboard */}
```
Comment (ryan-hamblin) [inline src/components/pages/Navbar/Navbar.js:129]: Excellent use of Code Comments.

File: src/components/pages/Navbar/NavbarFeatures/LoginButton.js [hunk 7]
```diff
@@ -1,9 +1,25 @@
...
+  useEffect(() => {
+    const getUserData = async () => {
+      try {
+        const token = await getAccessTokenSilently({
+          audience: 'underdog-identifier',
+          scope: 'openid profile email',
+        });
+        localStorage.setItem('AuthToken', token);
+      } catch (e) {
+        console.log(e.message);
```
Comment (ryan-hamblin) [inline src/components/pages/Navbar/NavbarFeatures/LoginButton.js:18]: let's handle the error here.

File: src/components/pages/Navbar/NavbarFeatures/LogoutButton.js [hunk 8]
```diff
@@ -1,16 +1,20 @@
 import React from 'react';
 import { useAuth0 } from '@auth0/auth0-react';
 
+import { NavBtnLink } from '../../NavBarLanding/NavBarStyle';
+
 function LogoutButton() {
   const { logout } = useAuth0();
 
+  const logoutAuth = () => {
```
Comment (ryan-hamblin) [inline src/components/pages/Navbar/NavbarFeatures/LogoutButton.js:9]: It looks like this is still WIP. We need to route 'somewhere'? I'd love to learn more about what the `logout()` fn does.

Comment (ryan-hamblin) [reply src/components/pages/Navbar/NavbarFeatures/LogoutButton.js:9]: @paulstgermain it sounds like you've worked this out and I know that  @miguelaledesma has worked on it a bit as well. Would love to know how this logout button works.

File: src/components/pages/Navbar/NavbarFeatures/SignupButton.js [hunk 9]
```diff
@@ -1,9 +1,25 @@
...
+  useEffect(() => {
+    const getUserData = async () => {
+      try {
+        const token = await getAccessTokenSilently({
+          audience: 'underdog-identifier',
+          scope: 'openid profile email',
+        });
+        localStorage.setItem('AuthToken', token);
+      } catch (e) {
+        console.log(e.message);
```
Comment (ryan-hamblin) [inline src/components/pages/Navbar/NavbarFeatures/SignupButton.js:18]: please handle this error.

File: src/components/pages/ProfileList/ProfileListContainer.js [hunk 10]
```diff
@@ -9,8 +8,8 @@ import RenderProfileListPage from './RenderProfileListPage';
 
 // Here is an example of using our reusable List component to display some list data to the UI.
 const ProfileList = () => {
-  const { authState } = useOktaAuth();
-
+  //Variable is created as place-holder. Function destroyed while removing Okta
```
Comment (ryan-hamblin) [inline src/components/pages/ProfileList/ProfileListContainer.js:11]: I think this is a typo. you need a space after `//` in order for the compiler to see it as commented out.

File: src/components/pages/ProfileList/ProfileListContainer.js [hunk 11]
```diff
@@ -9,8 +8,8 @@ import RenderProfileListPage from './RenderProfileListPage';
...
+  const authState = null;
```
Comment (ryan-hamblin) [inline src/components/pages/ProfileList/ProfileListContainer.js:12]: something is off here possibly w/ spaces/tabs? Not sure but the compiler doesn't like it.

File: src/index.js [hunk 12]
```diff
@@ -44,12 +40,13 @@ import { Provider } from 'react-redux';
...
 import thunk from 'redux-thunk';
-import { Auth0Provider } from '@auth0/auth0-react';
+import Auth0ProviderWithHistory from './auth/auth0ProviderWithHistory';
+// import { Auth0Provider } from '@auth0/auth0-react';
 
 import PrivateRoute from './components/common/PrivateRoute';
 
-const domain = process.env.REACT_APP_AUTH0_DOMAIN;
-const clientId = process.env.REACT_APP_AUTH0_CLIENT_ID;
+// const domain = process.env.REACT_APP_AUTH0_DOMAIN;
```
Comment (ryan-hamblin) [inline src/index.js:48]: take a peek here. Might need some spacing/formatting?

**Review by ryan-hamblin — CHANGES_REQUESTED** [review]
First things first. This is what a giant refactor or 'find and replace' style task looks like. So the work here is top notch! There were a few problems as stated throughout my review. Simplistic style tweaks to formatting and some lingering console.logs etc. Please let me know if you have any questions, feel free to reply directly inline with my review comments.

run formatter across the whole app. Lots of linter warnings will go away as a result. 


Problems found in `src/components/common/PrivateRoute.js` and `/src/__tests__/MenteeDasbhoard.test.js`

Remove lines 28 and 30 in src/components/common/SideBar/SideBar.js

File: src/components/pages/Navbar/NavbarFeatures/LoginButton.js (hunk 7 above)
Comment (mcfrr3) [reply src/components/pages/Navbar/NavbarFeatures/LoginButton.js:18]: Could we meet to discuss what you had in mind?

**Review by nathancai97 — APPROVED** [review]
Great job finding all the Okta related code and I agree with Melody, the four unused components with Okta should be grouped together/commented unless we actually have a use case for it. I am excited to see what comes next with our transition from Okta to Auth0!

**Review by nathancai97 — APPROVED** [review]
Great job finding all the Okta related code updating the package.json, and I agree with Melody, the four unused components with Okta should be grouped together/commented unless we actually have a use case for it. I am excited to see what comes next with our transition from Okta to Auth0!

File: src/components/common/PrivateRoute.js [hunk 13]
```diff
@@ -1,67 +1,19 @@
...
-    dispatch,
-    oktaAuth,
-    authState,
-    push,
-  ]);
+const PrivateRoute = ({ component, path, ...args }) => (
+  <Sidebar>
+    <Route
+      path={path}
+      /* NOTE: this is commented out for current functionality, but is a template for building auth0 into the app*/
```
Comment (Charles-Merrill-Johnson) [inline src/components/common/PrivateRoute.js:10]: Great use of code comments. Not only informational but also showing the current use and future implementation is fantastic. Good job!

**Review by Charles-Merrill-Johnson — COMMENTED** [review]
Great job on the fix. Removing the PrivateRoute to confirm Auth0 was indeed working was the right choice. This method will allow Auth0 to be implemented from the ground up!

**Review by jettyblue — APPROVED** [review]
Wow, I know this was a monster of a project but you all did it! Besides the little tweaks mentioned by Melody and Ryan, very nice work done here!

**Review by Charles-Merrill-Johnson — APPROVED** [review]
Great job on the fix. Removing the PrivateRoute to confirm Auth0 was indeed working was the right choice. This method will allow Auth0 to be implemented from the ground up!

**Review by zdravoj — APPROVED** [review]
Great work keeping track of a huge project like this. Everything looks good. Also, nice job correcting those styling inaccuracies in MenteeDashboard.test.js.

**Review by khaleel737 — APPROVED** [review]
Great job with implementing Auth0 and removing Okta, there is only this section [src/components/pages/PendingApproval/PendingApproval.js] where the automated message is hardcoded and does not reflect the name of the user, also the handlelogout doesnt remove Auth0 from localstorage, overall it looks great

**Review by MelodyLayne — APPROVED** [review]

**Review by brimjustin — APPROVED** [review]
Great work! The switch to Auth0 will allow for better integration, as well as your assignee changes.

**Review by ryan-hamblin — DISMISSED** [review]
looks like all my qualms were resolved.

**Pull request merged**